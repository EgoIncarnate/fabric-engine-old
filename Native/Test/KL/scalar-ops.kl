operator entry()
{
  var Scalar x;

  // Unary ops;
  x = 4.5;
  report +x;
  report -x;
  report !x;

  // Binary ops
  x = 1.2 + 5.6; report x;
  x = -5.2 - 9.0; report x;
  x = 9.4 * 1.2; report x;
  x = -9.3 / 5.4; report x;
  x = 8.9 % 2.1; report x;

  // Assignment ops
  x = 7.1; report x;
  x += 2.1; report x;
  x -= -4.7; report x;
  x *= 4.5; report x;
  x /= -1.2; report x;
  x %= 3.4; report x;

  // Intrinsics and build-ins
  report abs( -1.45 );
  report abs( 1.45 );
  report abs( -0.0 );
  report abs( 0.0 );
  report sin( 5.72283 );
  report cos( 5.72283 );
  report sqrt( 5.72283 );
  report pow( 5.72283, 3.1452 );
}
