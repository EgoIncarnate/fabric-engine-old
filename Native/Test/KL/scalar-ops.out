4.5
-4.5
false
6.8
-14.2
11.28
-1.72222
0.5
7.1
9.2
13.9
62.55
-52.125
-1.125
1.45
1.45
0
0
-0.531487
0.847066
2.39224
241.456
